### PR TITLE
Add GitHub org support

### PR DESCRIPTION
**Pull request by milescrabill**
@jvehent r+?

### CODE REVIEW

File: modules/github/github.go [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+	ldapToGithub map[string]string
+}
+
+type organization struct {
+	Name  string
+	Teams []string
+}
+
+type parameters struct {
+	Organizations    []organization
```
Comment (jvehent) [inline modules/github/github.go]: Is it wise to support several organizations at the same time, or should we limit one userplex module conf to one organization? My gut feeling is the later would cover the use case and avoid configuration mistakes.

Comment (milescrabill) [reply modules/github/github.go]: Yep, fixed to limit to one organization per module config.

File: modules/github/github.go [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+	membersMap = make(map[string]bool)
+	opt := &github.OrganizationListTeamMembersOptions{
+		ListOptions: github.ListOptions{
+			PerPage: 100,
+		},
+	}
+	for {
+		members, resp, err := r.ghclient.Organizations.ListTeamMembers(*team.ID, opt)
+		if err != nil || resp.StatusCode != 200 {
+			log.Printf("[error] github: could not list members for organization %s, error: %v with status %s", team, err, resp.Status)
```
Comment (jvehent) [inline modules/github/github.go:259]: shouldn't the function exit here?

File: modules/github/github.go [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+
+func (r *run) getOrgTeamsMap(org organization) (teamsMap map[string]*github.Team) {
+	teamsMap = make(map[string]*github.Team)
+	opt := &github.ListOptions{
+		PerPage: 100,
+	}
+	for {
+		teams, resp, err := r.ghclient.Organizations.ListTeams(org.Name, opt)
+		if err != nil || resp.StatusCode != 200 {
+			log.Printf("[error] github: could not list teams for organization %s, error: %v", org.Name, err)
```
Comment (jvehent) [inline modules/github/github.go:280]: same thing, shouldn't the function exit?

File: modules/github/github.go [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+			// if the team in config doesn't exist on github
+			if team, ok := teamsMap[teamName]; !ok {
+				log.Printf("[error] github: could not find team %s for organization %s", team, r.p.Organization.Name)
+			} else {
+				// if the user is already in the team, skip adding them
+				if _, ok := teamMembersMap[teamName][user]; ok {
+					continue
+				}
+				// user not in team, add them
+				if r.Conf.ApplyChanges && r.Conf.Create {
```
Comment (jvehent) [inline modules/github/github.go:132]: If not `ApplyChanges`, log that the user would have been added to the team

Comment (jvehent) [inline modules/github/github.go:132]: Nevermind, it's done later.

**Comment by jvehent** [comment]
Looks good, I'm merging it and you can address the minor comments in a follow up.

**Pull request merged**